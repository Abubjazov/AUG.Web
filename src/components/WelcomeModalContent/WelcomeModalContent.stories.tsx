import type { Meta, StoryObj } from '@storybook/react'
import MockedProvider from 'mockData/mockedReduxProvider'

import WelcomeModalContent, {
  WelcomeModalContentProps,
} from './WelcomeModalContent'

import '/src/index.css'

const meta: Meta<WelcomeModalContentProps> = {
  component: WelcomeModalContent,
  title: 'COMPONENTS/WelcomeModalContent',
  parameters: {
    docs: {
      description: {
        component: 'Interface component - WelcomeModalContent',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    NoProps: {
      description: 'There are no props for this component',
    },
  },
}

export default meta

type Story = StoryObj<WelcomeModalContentProps>

export const Default: Story = {
  args: {},

  decorators: [
    Story => (
      <MockedProvider>
        <div
          style={{
            maxWidth: '300px',
            background: '#999999',
            padding: 30,
            borderRadius: 8,
          }}
        >
          <Story />
        </div>
      </MockedProvider>
    ),
  ],
}
