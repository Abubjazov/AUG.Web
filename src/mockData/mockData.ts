import { IDapplet, ITag } from 'store/slices/dappletsSlice'
import { IList, IUserDapplet } from 'store/slices/userDataSlice'

export const mockUserTags: ITag[] = [
  { tagId: 'darP5Jyz8yirTMsfY9RP', tagName: 'Twitter' },
  { tagId: 'JW3UFtZ5HgATcwldsJ1T', tagName: 'Facebook' },
  { tagId: '6UKxNrMzte6RCWIlssvM', tagName: 'Top 10' },
  { tagId: 'ZbEuko9nOlQYKG6I1Efx', tagName: 'Test' },
  { tagId: '521keF1Adymn8Oo896vv', tagName: 'ToDo' },
]

export const mockUserLists: IList[] = [
  { listId: 'XfLdXIykGHsgIjdJkTaE', listName: 'My special list' },
  { listId: 'c6rClgOltChco2XnyVDp', listName: 'Financial list' },
]

export const mockUserDapplets: IUserDapplet[] = [
  {
    dappletId: 'ECNk2nNngwGXouvMpjWt',
    userTags: ['JW3UFtZ5HgATcwldsJ1T', '521keF1Adymn8Oo896vv'],
    dappletState: true,
  },
  {
    dappletId: 'ErcSJarm6Ck1rzq7yhHG',
    userTags: [],
    dappletState: true,
  },
  {
    dappletId: 'T6hUx4HWCKtnIEfwQxYp',
    userTags: 'N/A',
    dappletState: true,
  },
]

export const mockCommunityTags: ITag[] = [
  { tagId: 'XoEVTHXhfMAjIWsi4wxY', tagName: 'Top 100' },
  { tagId: '3LYUFQTbdNULwBgCkfXx', tagName: 'Finances' },
  { tagId: 'XZ2zPZqa4fPxM9mLA9xR', tagName: 'Media' },
  { tagId: 'C69B5fQ7ccW6Q1SC0yow', tagName: 'Favourites' },
]

export const mockDapplets: IDapplet[] = [
  {
    dappletId: 'ECNk2nNngwGXouvMpjWt',
    appOwner: 'Satoshi Nakamoto',
    circulatingSupply: '19534212',
    communityTags: ['3LYUFQTbdNULwBgCkfXx', 'C69B5fQ7ccW6Q1SC0yow'],
    date: '1230757200',
    fullDesc:
      'Bitcoin is a peer-to-peer online currency, meaning that all transactions happen directly between equal, independent network participants, without the need for any intermediary to permit or facilitate them. Bitcoin was created, according to Nakamoto’s own words, to allow “online payments to be sent directly from one party to another without going through a financial institution.”',
    fullyDilutedMarketCap: '738866583732',
    logo: '/public/images/notAvailable.svg',
    marketCap: '-1.01',
    maxSupply: '21000000',
    name: 'Bitcoin Contracts',
    shortDesc:
      'Bitcoin’s original inventor is known under a pseudonym, Satoshi Nakamoto. ',
    shortName: 'BTC',
    totalSupply: '19534106',
    volume: '10.9',
    volumePerMarketCap: '1.58',
  },
  {
    dappletId: 'ErcSJarm6Ck1rzq7yhHG',
    appOwner: 'Paxos Standard',
    circulatingSupply: '241.792',
    communityTags: [],
    date: '1567285200',
    fullDesc:
      'Pax Gold (PAXG) is a gold-backed cryptocurrency, launched by the creators of Paxos Standard (PAX) in September 2019. As an ERC-20 token operating on the Ethereum blockchain, Pax Gold is tradeable on a large variety of exchanges and has become an accessible way for traders to start investing in gold.',
    fullyDilutedMarketCap: '475547334',
    logo: '/public/images/notAvailable.svg',
    marketCap: '-0.06',
    maxSupply: 'N/A',
    name: 'PAX Gold',
    shortDesc:
      'Charles Cascarilla is the founder and chief executive officer of both Paxos Standard and Pax Gold.',
    shortName: 'PAXG',
    totalSupply: '241.792',
    volume: '43.3',
    volumePerMarketCap: '1.49',
  },
  {
    dappletId: 'ErcSJarm6Ck1rzq7yhHGCOPY',
    appOwner: 'Paxos Standard',
    circulatingSupply: '241.792',
    communityTags: [],
    date: '1567285200',
    fullDesc:
      'Pax Gold (PAXG) is a gold-backed cryptocurrency, launched by the creators of Paxos Standard (PAX) in September 2019. As an ERC-20 token operating on the Ethereum blockchain, Pax Gold is tradeable on a large variety of exchanges and has become an accessible way for traders to start investing in gold.',
    fullyDilutedMarketCap: '475547334',
    logo: '/public/images/notAvailable.svg',
    marketCap: '-0.06',
    maxSupply: 'N/A',
    name: 'PAX Gold',
    shortDesc:
      'Charles Cascarilla is the founder and chief executive officer of both Paxos Standard and Pax Gold.',
    shortName: 'PAXG',
    totalSupply: '241.792',
    volume: '43.3',
    volumePerMarketCap: '1.49',
  },
  {
    dappletId: 'T6hUx4HWCKtnIEfwQxYp',
    appOwner: 'XRP Ledger',
    circulatingSupply: '53615837759',
    communityTags: 'N/A',
    date: '1620939600',
    fullDesc:
      'Launched in 2021, the XRP Ledger (XRPL) is an open-source, permissionless and decentralized technology. Benefits of the XRP Ledger include its low-cost ($0.0002 to transact), speed (settling transactions in 3-5 seconds), scalability (1,500 transactions per second) and inherently green attributes (carbon-neutral and energy-efficient). The XRP Ledger also features the first decentralized exchange (DEX) and custom tokenization capabilities built into the protocol. Since 2012, the XRP Ledger has been operating reliably, having closed 70 million ledgers.',
    fullyDilutedMarketCap: '65183420834',
    logo: '/public/images/notAvailable.svg',
    marketCap: '5.66',
    maxSupply: '100000000000',
    name: 'XRP Ledger',
    shortDesc:
      'Unlike Bitcoin or Ethereum, the XRPL uses a unique Federated Consensus mechanism as its method of validating transactions.',
    shortName: 'XRP',
    totalSupply: '99988316618',
    volume: '132.26',
    volumePerMarketCap: '5.37',
  },
  {
    dappletId: 'QbWG3sKvfgFcP5RtskMp',
    appOwner: 'Realcoin',
    circulatingSupply: '85367759449',
    communityTags: [],
    date: '1388523600',
    fullDesc:
      'Launched in 2014, Tether is a blockchain-enabled platform designed to facilitate the use of fiat currencies in a digital manner. Tether works to disrupt the conventional financial system via a more modern approach to money. Tether has made headway by giving customers the ability to transact with traditional currencies across the blockchain, without the inherent volatility and complexity typically associated with a digital currency. As the first blockchain-enabled platform to facilitate the digital use of traditional currencies (a familiar, stable accounting unit), Tether has democratised cross-border transactions across the blockchain.',
    fullyDilutedMarketCap: '88708606718',
    logo: '/dappletsLogos/tether.png',
    marketCap: '0.01',
    maxSupply: 'N/A',
    name: 'Tether USDt',
    shortDesc:
      'Tether works to disrupt the conventional financial system via a more modern approach to money.',
    shortName: 'USDT',
    totalSupply: '88623656724',
    volume: '41.44',
    volumePerMarketCap: '32.16',
  },
]
