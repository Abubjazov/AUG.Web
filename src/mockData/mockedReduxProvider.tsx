import { PropsWithChildren } from 'react'

import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'

import {
  mockCommunityTags,
  mockDapplets,
  mockMyDapplets,
  mockMyTags,
} from './mockData'
import dappletsReducer from '../store/slices/dappletsSlice'
import layoutReducer from '../store/slices/layoutSlice'
import myDappletsReducer from '../store/slices/myDappletsSlice'

export const defaultMockState = {
  layout: {
    menuOpened: false,
    dappletSettingsOpened: false,
  },

  myDapplets: {
    myDapplets: [...mockMyDapplets],
    myTags: [...mockMyTags],
  },

  dapplets: {
    dapplets: [...mockDapplets],
    tags: [...mockCommunityTags],
  },
}

export const mockedReduxProvider = ({
  children,
}: PropsWithChildren<object>) => {
  const store = configureStore({
    reducer: {
      layout: layoutReducer,
      dapplets: dappletsReducer,
      myDapplets: myDappletsReducer,
    },
    preloadedState: {
      ...defaultMockState,
    },
  })

  return <Provider store={store}>{children}</Provider>
}
