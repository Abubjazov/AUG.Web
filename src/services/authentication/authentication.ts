import { createAsyncThunk } from '@reduxjs/toolkit'
import {
  createUserWithEmailAndPassword,
  getAuth,
  signInWithEmailAndPassword,
  signOut,
} from 'firebase/auth'
import { doc, getFirestore, setDoc } from 'firebase/firestore'
import { ISignUpData } from 'store/slices/authSlice'
import { EMessageType, addMessage } from 'store/slices/layoutSlice'
import { getErrorMessage } from 'utils/getErrorMessage/getErrorMessage'

export const createUser = createAsyncThunk(
  'auth/createUser',
  async (data: ISignUpData, { dispatch }) => {
    const auth = getAuth()
    const db = getFirestore()

    try {
      await createUserWithEmailAndPassword(
        auth,
        data.email,
        data.password,
      ).then(async user => {
        await setDoc(doc(db, 'UsersData', user.user.uid), {
          userDapplets: [],
          userTags: [],
          userLists: [],
        })
      })
    } catch (error) {
      dispatch(
        addMessage({
          messageText: getErrorMessage(error),
          messageType: EMessageType.ERROR,
        }),
      )
    }
  },
)

export const logIn = createAsyncThunk(
  'auth/logIn',
  async (data: ISignUpData, { dispatch }) => {
    const auth = getAuth()

    try {
      await signInWithEmailAndPassword(auth, data.email, data.password)
    } catch (error) {
      dispatch(
        addMessage({
          messageText: getErrorMessage(error),
          messageType: EMessageType.ERROR,
        }),
      )
    }
  },
)

export const logOut = createAsyncThunk(
  'auth/logOut',
  async (_, { dispatch }) => {
    const auth = getAuth()

    try {
      await signOut(auth)
    } catch (error) {
      dispatch(
        addMessage({
          messageText: getErrorMessage(error),
          messageType: EMessageType.ERROR,
        }),
      )
    }
  },
)
