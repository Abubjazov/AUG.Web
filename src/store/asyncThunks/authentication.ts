import { createAsyncThunk } from '@reduxjs/toolkit'
import {
  apiCreateUser,
  apiLogIn,
  apiLogOut,
} from 'api/fireStore/fireStoreMethods'
import { ISignUpData, setIsInProgress } from 'store/slices/authSlice'
import { EMessageType, addMessage } from 'store/slices/layoutSlice'
import { getErrorMessage } from 'utils/getErrorMessage/getErrorMessage'

export const createUser = createAsyncThunk(
  'auth/createUser',
  async (data: ISignUpData, { dispatch }) => {
    try {
      dispatch(setIsInProgress(true))

      await apiCreateUser(data)
    } catch (error) {
      dispatch(
        addMessage({
          messageText: getErrorMessage(error),
          messageType: EMessageType.ERROR,
        }),
      )
    } finally {
      dispatch(setIsInProgress(false))
    }
  },
)

export const logIn = createAsyncThunk(
  'auth/logIn',
  async (data: ISignUpData, { dispatch }) => {
    try {
      dispatch(setIsInProgress(true))

      await apiLogIn(data)
    } catch (error) {
      dispatch(
        addMessage({
          messageText: getErrorMessage(error),
          messageType: EMessageType.ERROR,
        }),
      )
    } finally {
      dispatch(setIsInProgress(false))
    }
  },
)

export const logOut = createAsyncThunk(
  'auth/logOut',
  async (_, { dispatch }) => {
    try {
      dispatch(setIsInProgress(true))

      await apiLogOut()
    } catch (error) {
      dispatch(
        addMessage({
          messageText: getErrorMessage(error),
          messageType: EMessageType.ERROR,
        }),
      )
    } finally {
      dispatch(setIsInProgress(false))
    }
  },
)
