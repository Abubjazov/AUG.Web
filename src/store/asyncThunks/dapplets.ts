import { createAsyncThunk } from '@reduxjs/toolkit'
import {
  apiGetCommunityTags,
  apiGetDapplets,
} from 'api/fireStore/fireStoreMethods'
import {
  ITag,
  IWhere,
  TLastVisible,
  setDapplets,
  setIsLoadingDapplets,
  setIsLoadingMoreDapplets,
  setTags,
} from 'store/slices/dappletsSlice'
import { EMessageType, addMessage } from 'store/slices/layoutSlice'
import { getErrorMessage } from 'utils/getErrorMessage/getErrorMessage'

export interface IGetDappletOptions {
  withLimit?: number
  withStartAfter?: TLastVisible
  withWhere?: IWhere
}

export const getDapplets = createAsyncThunk<void, IGetDappletOptions>(
  'dapplets/getDapplets',
  async ({ withLimit, withStartAfter, withWhere }, { dispatch }) => {
    try {
      withStartAfter
        ? dispatch(setIsLoadingMoreDapplets(true))
        : dispatch(setIsLoadingDapplets(true))

      if (withWhere && !withWhere?.comparisonValue.length) {
        dispatch(
          setDapplets({
            dapplets: [],
            lastVisible: undefined,
            add: false,
          }),
        )

        return
      }

      const { dapplets, lastVisible } = await apiGetDapplets({
        withLimit,
        withStartAfter,
        withWhere,
      })

      dispatch(
        setDapplets({
          dapplets,
          lastVisible,
          add: Boolean(withStartAfter),
        }),
      )
    } catch (error) {
      dispatch(
        addMessage({
          messageText: getErrorMessage(error),
          messageType: EMessageType.ERROR,
        }),
      )
    } finally {
      withStartAfter
        ? dispatch(setIsLoadingMoreDapplets(false))
        : dispatch(setIsLoadingDapplets(false))
    }
  },
)

export const getCommunityTags = createAsyncThunk<void, never>(
  'dapplets/getCommunityTags',
  async (_, { dispatch }) => {
    try {
      const tags: ITag[] = await apiGetCommunityTags()

      dispatch(setTags(tags))
    } catch (error) {
      dispatch(
        addMessage({
          messageText: getErrorMessage(error),
          messageType: EMessageType.ERROR,
        }),
      )
    }
  },
)
