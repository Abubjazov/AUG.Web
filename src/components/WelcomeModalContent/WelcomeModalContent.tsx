import { FC, useState } from 'react'

import { ERenderMode } from 'components/AddUserTagModalContent/AddUserTagModalContent'
import SingInForm from 'components/SingInForm'
import SingUpForm from 'components/SingUpForm'
import { createUser, logIn } from 'store/asyncThunks/authentication'
import { useAppDispatch } from 'store/hooks'
import { ISignUpData } from 'store/slices/authSlice'
import BaseButton from 'uikit/BaseButton'
import { EBaseButtonMode } from 'uikit/BaseButton/BaseButton'

import styles from './WelcomeModalContent.module.css'

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface WelcomeModalContentProps {}

const WelcomeModalContent: FC<WelcomeModalContentProps> = () => {
  const dispatch = useAppDispatch()

  const [renderMode, setRenderMode] = useState<ERenderMode>(ERenderMode.WELCOME)

  const setRenderModeToWelcome = () => {
    setRenderMode(ERenderMode.WELCOME)
  }

  const onSignUp = (data: ISignUpData) => {
    void dispatch(createUser(data))
  }

  const onSignIn = (data: ISignUpData) => {
    void dispatch(logIn(data))
  }

  const renderWelcome = () => (
    <div className={styles['root']}>
      <span className={styles.title}>Welcome</span>

      <span className={styles.text}>
        We are building an Augmented Web platform consisting of a browser-based
        plugin and decentralized applications (dapplets) based on crypto
        technologies.
      </span>

      <span className={styles.text}>
        Our platform is created on an open-source basis and is available to
        developers from all over the world.
      </span>

      <div className={styles.buttons}>
        <BaseButton
          userStyles={styles.button}
          label={'Sign up'}
          mode={EBaseButtonMode.CONTAINED_RED}
          onClick={() => setRenderMode(ERenderMode.REGISTRATION)}
        />

        <BaseButton
          userStyles={styles.button}
          label={'Sign in'}
          onClick={() => setRenderMode(ERenderMode.LOGIN)}
        />
      </div>
    </div>
  )

  return (
    <>
      {renderMode === ERenderMode.WELCOME && renderWelcome()}

      {renderMode === ERenderMode.LOGIN && (
        <SingInForm userFunction={setRenderModeToWelcome} onSignIn={onSignIn} />
      )}

      {renderMode === ERenderMode.REGISTRATION && (
        <SingUpForm userFunction={setRenderModeToWelcome} onSignUp={onSignUp} />
      )}
    </>
  )
}

export default WelcomeModalContent
