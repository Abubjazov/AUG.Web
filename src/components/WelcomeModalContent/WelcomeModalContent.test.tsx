import { fireEvent, render, screen } from '@testing-library/react'
import { mockedReduxProvider as Provider } from 'mockData/mockedReduxProvider'
import * as asyncActions from 'store/asyncThunks/authentication'

import WelcomeModalContent from './WelcomeModalContent'

describe('WelcomeModalContent', () => {
  test('should render WelcomeModalContent default', () => {
    const { asFragment } = render(
      <Provider>
        <WelcomeModalContent />
      </Provider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render WelcomeModalContent with "Sign up"', () => {
    const { asFragment } = render(
      <Provider>
        <WelcomeModalContent />
      </Provider>,
    )

    fireEvent.click(screen.getByText('Sign up'))

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render WelcomeModalContent with "Sign in"', () => {
    const { asFragment } = render(
      <Provider>
        <WelcomeModalContent />
      </Provider>,
    )

    fireEvent.click(screen.getByText('Sign in'))

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render WelcomeModalContent default after close "Sign in"', () => {
    const { asFragment } = render(
      <Provider>
        <WelcomeModalContent />
      </Provider>,
    )

    fireEvent.click(screen.getByText('Sign in'))

    expect(asFragment()).toMatchSnapshot()

    fireEvent.click(screen.getByText('Cancel'))

    expect(asFragment()).toMatchSnapshot()
  })

  test('should call createUser function', () => {
    const mockedCreateUser = vi
      .spyOn(asyncActions, 'createUser')
      .mockImplementation(() => vi.fn())

    render(
      <Provider>
        <WelcomeModalContent />
      </Provider>,
    )

    fireEvent.click(screen.getByText('Sign up'))

    fireEvent.change(screen.getByTestId('email-input'), {
      target: { value: 'email@example.com' },
    })

    fireEvent.change(screen.getByTestId('password-input'), {
      target: { value: '12345678' },
    })

    fireEvent.change(screen.getByTestId('confirm-password-input'), {
      target: { value: '12345678' },
    })

    fireEvent.click(screen.getByText('Sign up'))

    expect(mockedCreateUser).toHaveBeenCalledTimes(1)
    expect(mockedCreateUser).toHaveBeenCalledWith({
      email: 'email@example.com',
      password: '12345678',
    })
  })

  test('should call logIn function', () => {
    const mockedLogIn = vi
      .spyOn(asyncActions, 'logIn')
      .mockImplementation(() => vi.fn())

    render(
      <Provider>
        <WelcomeModalContent />
      </Provider>,
    )

    fireEvent.click(screen.getByText('Sign in'))

    fireEvent.change(screen.getByTestId('email-input'), {
      target: { value: 'email@example.com' },
    })

    fireEvent.change(screen.getByTestId('password-input'), {
      target: { value: '12345678' },
    })

    fireEvent.click(screen.getByText('Sign in'))

    expect(mockedLogIn).toHaveBeenCalledTimes(1)
    expect(mockedLogIn).toHaveBeenCalledWith({
      email: 'email@example.com',
      password: '12345678',
    })
  })
})
