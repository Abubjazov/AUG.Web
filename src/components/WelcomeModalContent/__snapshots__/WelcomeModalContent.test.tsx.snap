// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WelcomeModalContent > should render WelcomeModalContent default 1`] = `
<DocumentFragment>
  <div
    class="_root_b3f100"
  >
    <span
      class="_title_b3f100"
    >
      Welcome
    </span>
    <span
      class="_text_b3f100"
    >
      We are building an Augmented Web platform consisting of a browser-based plugin and decentralized applications (dapplets) based on crypto technologies.
    </span>
    <span
      class="_text_b3f100"
    >
      Our platform is created on an open-source basis and is available to developers from all over the world.
    </span>
    <div
      class="_buttons_b3f100"
    >
      <button
        class="_root_91d6ec _contained-red_91d6ec  _button_b3f100"
        data-testid="base-button"
        type="button"
      >
        Sign up
      </button>
      <button
        class="_root_91d6ec _outlined-white_91d6ec  _button_b3f100"
        data-testid="base-button"
        type="button"
      >
        Sign in
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`WelcomeModalContent > should render WelcomeModalContent default after close "Sign in" 1`] = `
<DocumentFragment>
  <div
    class="_root_eae5fe"
  >
    <span
      class="_title_eae5fe"
    >
      Sign 
      <span
        class="_red-text_eae5fe"
      >
        in
      </span>
    </span>
    <div
      class="_root_672eb8"
    >
      <input
        data-testid="email-input"
        name="email"
        placeholder="email@example.com"
        type="email"
        value=""
      />
    </div>
    <div
      class="_root_672eb8"
    >
      <input
        data-testid="password-input"
        name="password"
        placeholder="password"
        type="password"
        value=""
      />
    </div>
    <div
      class="_buttons_eae5fe"
    >
      <button
        class="_root_91d6ec _contained-red_91d6ec  _button_eae5fe"
        data-testid="base-button"
        type="button"
      >
        Sign in
      </button>
      <button
        class="_root_91d6ec _outlined-white_91d6ec  _button_eae5fe"
        data-testid="base-button"
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`WelcomeModalContent > should render WelcomeModalContent default after close "Sign in" 2`] = `
<DocumentFragment>
  <div
    class="_root_b3f100"
  >
    <span
      class="_title_b3f100"
    >
      Welcome
    </span>
    <span
      class="_text_b3f100"
    >
      We are building an Augmented Web platform consisting of a browser-based plugin and decentralized applications (dapplets) based on crypto technologies.
    </span>
    <span
      class="_text_b3f100"
    >
      Our platform is created on an open-source basis and is available to developers from all over the world.
    </span>
    <div
      class="_buttons_b3f100"
    >
      <button
        class="_root_91d6ec _contained-red_91d6ec  _button_b3f100"
        data-testid="base-button"
        type="button"
      >
        Sign up
      </button>
      <button
        class="_root_91d6ec _outlined-white_91d6ec  _button_b3f100"
        data-testid="base-button"
        type="button"
      >
        Sign in
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`WelcomeModalContent > should render WelcomeModalContent with "Sign in" 1`] = `
<DocumentFragment>
  <div
    class="_root_eae5fe"
  >
    <span
      class="_title_eae5fe"
    >
      Sign 
      <span
        class="_red-text_eae5fe"
      >
        in
      </span>
    </span>
    <div
      class="_root_672eb8"
    >
      <input
        data-testid="email-input"
        name="email"
        placeholder="email@example.com"
        type="email"
        value=""
      />
    </div>
    <div
      class="_root_672eb8"
    >
      <input
        data-testid="password-input"
        name="password"
        placeholder="password"
        type="password"
        value=""
      />
    </div>
    <div
      class="_buttons_eae5fe"
    >
      <button
        class="_root_91d6ec _contained-red_91d6ec  _button_eae5fe"
        data-testid="base-button"
        type="button"
      >
        Sign in
      </button>
      <button
        class="_root_91d6ec _outlined-white_91d6ec  _button_eae5fe"
        data-testid="base-button"
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`WelcomeModalContent > should render WelcomeModalContent with "Sign up" 1`] = `
<DocumentFragment>
  <div
    class="_root_702078"
  >
    <span
      class="_title_702078"
    >
      Sign 
      <span
        class="_red-text_702078"
      >
        up
      </span>
    </span>
    <div
      class="_root_672eb8"
    >
      <input
        data-testid="email-input"
        name="email"
        placeholder="email@example.com"
        type="email"
        value=""
      />
    </div>
    <div
      class="_root_672eb8"
    >
      <input
        data-testid="password-input"
        name="password"
        placeholder="password"
        type="password"
        value=""
      />
    </div>
    <div
      class="_root_672eb8"
    >
      <input
        data-testid="confirm-password-input"
        name="confirm-password"
        placeholder="password confirm"
        type="password"
        value=""
      />
    </div>
    <div
      class="_buttons_702078"
    >
      <button
        class="_root_91d6ec _contained-red_91d6ec  _button_702078"
        data-testid="base-button"
        type="button"
      >
        Sign up
      </button>
      <button
        class="_root_91d6ec _outlined-white_91d6ec  _button_702078"
        data-testid="base-button"
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
</DocumentFragment>
`;
